haystack/pipeline.py

```diff
@@ -66,6 +66,9 @@ def get_node(self, name: str):
         component = self.graph.nodes[name]["component"]
         return component
 
+    def update_node(self, name: str, component):
```

Comment:
1) Please add a short docstring for all methods 
2) naming: I am fine with update_node() but maybe `set_node` would be more intuitive given the already existing `get_node()`?